drivers/counter/counter_ll_stm32_timer.c

```diff
@@ -0,0 +1,623 @@
+/*
+ * Copyright (c) 2021 Kent Hall.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT st_stm32_counter
+
+#include <drivers/counter.h>
+#include <drivers/clock_control/stm32_clock_control.h>
+#include <stm32_ll_tim.h>
+#include <stm32_ll_rcc.h>
```

Comment:
nit: better include non-Zephyr headers after

Reply:
Okay done, though I've still kept it before `logging/log.h` since this seems to be kept last for other drivers.